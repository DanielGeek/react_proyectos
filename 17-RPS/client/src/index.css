body, html, div#root {
    background: #f93e17;
    font-family: sans-serif, 'Roboto Condensed';
}
.container {
    margin-top: 40px;
}