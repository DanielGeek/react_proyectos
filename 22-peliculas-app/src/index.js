import React from "react";
import ReactDOM from "react-dom";
import App from "./App";

import "bootstrap/dist/css/bootstrap.css";
import "bootstrap/dist/css/bootstrap-theme.css";

ReactDOM.render(
  <App nombre="Daniel" mensaje="Este es un mensaje" />,
  document.getElementById("root")
);
