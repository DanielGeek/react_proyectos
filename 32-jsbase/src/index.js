const cities = ['Caracas', 'Buenos Aires', 'Santiago', 'Ciudad de MÃ©xico'];

const randomString = () => {
  const strings = cities[Math.floor(Math.random() * cities.length)];
  return strings;
};

const reverseString2 = str => {
  return new Promise((resolve, reject) => {
    if (!str) {
      reject(Error('Error'));
    }
    resolve(str.split("").reverse().join(""));
  });
};

module.exports = randomString;
