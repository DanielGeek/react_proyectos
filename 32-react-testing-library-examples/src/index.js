import React from 'react';
import ReactDOM from 'react-dom';
import { createStore } from 'redux';
import { noteReducer } from './reducers/noteReducer';

const store = createStore(noteReducer);

store.dispatch({
  type: '@notes/created',
  payload: {
    content: 'I like the classes with Dani',
    important: true,
    id: 1
  }
})

store.dispatch({
  type: '@notes/created',
  payload: {
    content: 'But i need learn',
    important: false,
    id: 2
  }
})

const App = () => {
  const state = store.getState();

  return (
    <ul>
      {
        state.map(note => {
          return <li key={note.id}>
            {note.content}
            <strong>
              {
                note.important
                  ? 'important'
                  : 'not important'
              }
            </strong>
          </li>
        })
      }
    </ul>
  )
}

const renderApp = () => {
  ReactDOM.render(
    <App />,
    document.getElementById('root')
  );
}

renderApp()
store.subscribe(renderApp)