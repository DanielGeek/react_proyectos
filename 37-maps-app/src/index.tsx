import React from 'react';
import ReactDOM from 'react-dom';
import { MapsApp } from './MapsApp';

//@ts-ignore
// eslint-disable-next-line import/no-webpack-loader-syntax
import mapboxgl from '!mapbox-gl'; // or "const mapboxgl = require('mapbox-gl');"

mapboxgl.accessToken = 'pk.eyJ1IjoiZGFuaWVsZ2VlayIsImEiOiJja291eHRlODQwMm5hMm5sOXM3YzlwajlxIn0.UORg5svptSrVMXj_FSENAg';

if ( !navigator.geolocation ) {
  alert('You browser not have geolocation option');
  throw new Error('You browser not have geolocation option');
}

ReactDOM.render(
  <React.StrictMode>
    <MapsApp />
  </React.StrictMode>,
  document.getElementById('root')
);

