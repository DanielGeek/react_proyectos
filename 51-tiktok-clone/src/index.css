@font-face {
  font-family: 'ProximaNova';
  src: url('assets/Proxima-Nova-Bold.woff2')
  format('woff2');
  font-weight: 400;
}

@font-face {
  font-family: 'ProximaNova';
  src: url('assets/Proxima-Nova-Semibold.woff2')
  format('woff2');
  font-weight: 600;
}

:root {
  margin: 0;
  font-family: ProximaNova,Arial,Tahoma,PingFangSC,sans-serif;
  font-size: 16px;
  line-height: 24px;
  font-weight: 400;

  color-scheme: light dark;
  color: rgba(255, 255, 255, 0.87);
  background-color: #242424;

  font-synthesis: none;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  -webkit-text-size-adjust: 100%;
}

html {
  box-sizing: border-box;
}

*, *::before, *::after {
  box-sizing: inherit;
}


