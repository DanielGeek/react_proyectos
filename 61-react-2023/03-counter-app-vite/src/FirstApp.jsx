
export const FirstApp = () => {

    return (
        <>
            <h1>Daniel</h1>
            {/* <code>{JSON.stringify( newMessage )}</code> */}
            <p>Spy a subtitle</p>
        </>
    )
}
