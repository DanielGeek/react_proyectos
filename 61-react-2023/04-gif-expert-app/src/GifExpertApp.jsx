import { useState } from "react";
import { AddCategory } from "./components/AddCategory";

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch', 'Goku']);

    const onAddCategory = ( newCategory ) => {

        setCategories([ newCategory, ...categories ]);
        // setCategories( cat => [ ...cat, 'Valorant' ] );
    }

    return (
        <>
            <h1>GifExpertApp</h1>
            
            <AddCategory
                // setCategories={ setCategories }
                onNewCategory={ (value) => onAddCategory(value) }
            />

            {/* <button onClick={ onAddCategory }>Add</button> */}
            <ol>
                { categories.map( category => {
                    return <li key={ category }>{ category }</li>
                }) }
            </ol>
        </>
    );
};
