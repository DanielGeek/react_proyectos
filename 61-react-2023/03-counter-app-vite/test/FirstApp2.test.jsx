import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Testing in <FirstApp />', () => {

    const title = "Hello, I'm Goku";
    const subTitle = "I'm a subTitle";

    test('Must match snapshot', () => {
        
        const { container } = render( <FirstApp title={ title } /> );
        expect( container ).toMatchSnapshot();

    });

    test("Must show the message 'Hello, I'm Goku' ", () => {

        render( <FirstApp title={ title } /> );
        expect( screen.getByText(title) ).toBeTruthy();
        // screen.debug();

    });

    test('Must show title in h1', () => {
        render( <FirstApp title={ title } /> );
        expect( screen.getByRole('heading', { level: 1 }).innerHTML ).toContain( title );
    });

    test('Must show subTitle sent in props', () => {

        render(
            <FirstApp
                title={ title }
                subTitle={ subTitle }
            />
        );

        expect( screen.getAllByText(subTitle).length ).toBe(2);

    });

});