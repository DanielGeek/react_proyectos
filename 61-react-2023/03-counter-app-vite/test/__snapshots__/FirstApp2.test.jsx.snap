// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in <FirstApp /> Must match snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hello, I'm Goku
  </h1>
  <p>
    There is not subtitle
  </p>
  <p>
    There is not subtitle
  </p>
  <p>
    Daniel Angel
  </p>
</div>
`;
